lib/active_model/serializer.rb

```diff
@@ -177,6 +177,10 @@ def json_key
       end
     end
 
+    def as_json
+      self.class.adapter.new(self).as_json
```

Comment:
As we discussed earlier https://github.com/rails-api/active_model_serializers/pull/889#issuecomment-109348755 and in https://github.com/rails-api/active_model_serializers/pull/936#discussion_r31732511 I think the real issue is that the the code in the actioncontroller::serializer that wraps the model to be rendered in the adapter needs to be encapsulated as 'the way' a resource is serialized in one nice step.  I'm intended on making that PR sometime in the next week.